### ISSUE TITLE
session cookie not persisted on restart when applying browser update - follow up to 28379

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#29135]
## Description 

Found while testing #28379 

When session cookie is set and I go to apply browser update (used brave://settings/help), the session cookie is not retained after browser restart from that page.

Note - could be macOS specific as @goodov could not reproduce on Windows. Requesting others on @brave/qa-team to try various platforms to see if it reproduces for them.

## Steps to Reproduce

1. Install slightly old beta version which includes this fix, but which will want to update. (I used 1.50.86)
2. Ensure "Continue where you left off" is enabled.
3. Open https://bayden.com/test/sessions/ and choose `lightcyan` from selector.
4. Ensure `sessionCookie says color is: lightcyan`.
5. Navigate to brave://settings/help and apply update via "Relaunch" button, browser should restart automatically.
6. Expect the session is restored and the website is still opened.
7. Navigate to test page and refresh the page (Ctrl/Cmd+R).
--> Expect `sessionCookie` is still `lightcyan`. However `sessionCookie` field `has no value` instead

## Actual result:

<img width="1190" alt="8" src="https://user-images.githubusercontent.com/28145373/225917554-ad44686a-69c7-4c48-9ceb-1c1b39edb969.png">

## Expected result:

`sessionCookie` should be `lightcyan` after applying browser restart in this case

## Reproduces how often: 

easily on macOS, @goodov could not reproduce on Windows

## Brave version (brave://version info)

```
Brave | 1.50.91 Chromium: 111.0.5563.64 (Official Build) beta (x86_64)
-- | --
Revision | c710e93d5b63b7095afe8c2c17df34408078439d-refs/branch-heads/5563@{#995}
OS | macOS Version 12.6.3 (Build 21G419)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? n/a
- Can you reproduce this issue with the beta channel? yes
- Can you reproduce this issue with the nightly channel? likely

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @rebron @goodov @brave/qa-team

**Comment by goodov** [comment]
the underlying issue is in the update process: https://github.com/brave/brave-browser/issues/25576

**Comment by LaurenWags** [comment]
@goodov I did the following:

1. Launched 1.50.86 as follows `/Applications/Brave\ Browser\ Beta.app/Contents/MacOS/Brave\ Browser\ Beta --args --enable-logging=stderr --variations-server-url=https://variations.bravesoftware.com/seed` - confirm version on brave://version, leave this page open
2. Ensured "Continue where you left off" is enabled.
3. Open https://bayden.com/test/sessions/ and choose `lightcyan` from selector.
4. Ensured `sessionCookie says color is: lightcyan`.
5. Navigate to brave://settings/help and apply update via "Relaunch" button, browser should restart automatically.
6. Expect the session is restored and the website is still opened.
7. Navigate to brave://version tab and confirm `BraveCleanupSessionCookiesOnSessionRestore:Disabled` shows on brave://version
8. Navigate to test page and refresh the page (Ctrl/Cmd+R).
9. Confirm `sessionCookie` is still `lightcyan`. 

Step 1 | Step 2 | Step 4 | Step 5 | Step 7 | Step 9
---- | ---- | ---- | ---- | ---- | ----
<img width="1189" alt="1" src="https://user-images.githubusercontent.com/28145373/225956303-3c57ddf4-7578-4f41-8c1e-3f7330971b21.png"> | <img width="1192" alt="2" src="https://user-images.githubusercontent.com/28145373/225956306-91a0c0ce-e3ce-439f-a4b6-ce3bb0f3fc8f.png"> | <img width="1188" alt="4" src="https://user-images.githubusercontent.com/28145373/225956308-9813a274-ebbb-4b81-9b97-04ea928f3aad.png"> | <img width="1189" alt="5" src="https://user-images.githubusercontent.com/28145373/225956311-f47de144-cffc-4897-873a-b6e0c36fa50e.png"> | <img width="1190" alt="7" src="https://user-images.githubusercontent.com/28145373/225956312-86544fc3-59c4-4f47-8a38-754bb5d50cc9.png"> | <img width="1184" alt="9" src="https://user-images.githubusercontent.com/28145373/225956315-c9f527b9-d3b8-419a-a229-88760b9f8340.png">

**Comment by MadhaviSeelam** [comment]
Followed STR from https://github.com/brave/brave-browser/issues/29135#issue-1629322750 and couldn't reproduce in Win 11 x64 

cc: @LaurenWags @goodov 


https://user-images.githubusercontent.com/98358127/225956056-99ff48c2-00b9-456e-bfe1-779c19369072.mp4

**Comment by goodov** [comment]
> Followed STR from [#29135 (comment)](https://github.com/brave/brave-browser/issues/29135#issue-1629322750) and couldn't reproduce in Win 11 x64
> 
> cc: @LaurenWags @goodov

yes, thank you! That's expected. We have this issue only on MacOS because of the broken manual update process.

**Comment by btlechowski** [comment]
Unable to reproduce on Linux

Brave | 1.50.93 Chromium: 111.0.5563.64 (Official Build) beta (64-bit)
-- | --
Revision | c710e93d5b63b7095afe8c2c17df34408078439d-refs/branch-heads/5563@{#995}
OS | Ubuntu 18.04 LTS


![image](https://user-images.githubusercontent.com/34715963/225973893-6834cc13-cf1e-4fc8-b700-d539b47d607a.png)

**Comment by kjozwiak** [comment]
> @goodov I did the following:
> 
> 1. Launched 1.50.86 as follows `/Applications/Brave\ Browser\ Beta.app/Contents/MacOS/Brave\ Browser\ Beta --args --enable-logging=stderr --variations-server-url=https://variations.bravesoftware.com/seed` - confirm version on brave://version, leave this page open
> 2. Ensured "Continue where you left off" is enabled.
> 3. Open https://bayden.com/test/sessions/ and choose `lightcyan` from selector.
> 4. Ensured `sessionCookie says color is: lightcyan`.
> 5. Navigate to brave://settings/help and apply update via "Relaunch" button, browser should restart automatically.
> 6. Expect the session is restored and the website is still opened.
> 7. Navigate to brave://version tab and confirm `BraveCleanupSessionCookiesOnSessionRestore:Disabled` shows on brave://version
> 8. Navigate to test page and refresh the page (Ctrl/Cmd+R).
> 9. Confirm `sessionCookie` is still `lightcyan`.
> 
> Step 1	Step 2	Step 4	Step 5	Step 7	Step 9
> <img alt="1" width="1189" src="https://user-images.githubusercontent.com/28145373/225956303-3c57ddf4-7578-4f41-8c1e-3f7330971b21.png">	<img alt="2" width="1192" src="https://user-images.githubusercontent.com/28145373/225956306-91a0c0ce-e3ce-439f-a4b6-ce3bb0f3fc8f.png">	<img alt="4" width="1188" src="https://user-images.githubusercontent.com/28145373/225956308-9813a274-ebbb-4b81-9b97-04ea928f3aad.png">	<img alt="5" width="1189" src="https://user-images.githubusercontent.com/28145373/225956311-f47de144-cffc-4897-873a-b6e0c36fa50e.png">	<img alt="7" width="1190" src="https://user-images.githubusercontent.com/28145373/225956312-86544fc3-59c4-4f47-8a38-754bb5d50cc9.png">	<img alt="9" width="1184" src="https://user-images.githubusercontent.com/28145373/225956315-c9f527b9-d3b8-419a-a229-88760b9f8340.png">

Verified the changes via Griffin using the template from above as per https://github.com/brave/brave-variations/pull/557#issuecomment-1486160746 before merging the change into `production` using https://github.com/brave/brave-variations/pull/558.

**Comment by rtritto** [comment]
Any update?
After restarting Brave (v1.65.132), cookies are empty.
